### DISCUSSION TITLE
Is there a way to make naudio/NAudio.WaveFormRenderer to work in UWP?

### DISCUSSION THREAD

**Discussion by Billeclipse** [naudio/NAudio#1030] [Q&A] [answered]
I see that only WasapiOutRT and WasapiCaptureRT exists for UWP and i know that the UWP support in NAudio is still in a beta state but is there any way to support naudio/NAudio.WaveFormRenderer and create waveforms for an UWP app?

**Comment by Panda-Sharp** [comment, marked as answer]
Now I'm out and on my phone, take a look here https://github.com/Yugen-Apps/Yugen.Toolkit/blob/main/Yugen.Toolkit.Uwp.Samples/ViewModels/Yugen/Audio/WaveformViewModel.cs you should find that answer, if you still have issues feel free to ping me in 10 days

**Reply by Billeclipse** [reply]
Thank you for your answer! I tried implementing the WaveformViewModel.cs that you suggested but i am getting a weird problem with it... I am able to load the waveform with the NaudioGenerateAudioData but if try to load more than once i get System.IO.IOException in NAudio.Wasapi.dll (I/O error occurred). Have you seen this error before? If yes how did you solve it?

**Reply by Panda-Sharp** [reply]
Hi @Billeclipse , I didn't noticed you edited your reply :) so I was focused to write a super simple working sample, here it is
https://github.com/Yugen-Apps/Yugen.Toolkit/tree/feature/waveform-sample/WaveFormSample.Uwp
please give it a shot and see if you can reproduce your issue, I can't!

P.s.
Be sure to use Naudio.* packages, the full `Naudio` doesn't support UWP (or at least only in part)

**Reply by Billeclipse** [reply]
Thank you very much again @Panda-Sharp for your help! I solve a lot of the first issues that i edited out of my original response by importing the correct libraries for ManagedBass and NAudio. I wished that NAudio will be fully supported for UWP in the future so that i can have some more options for my waveform. I noticed that my problem with the NaudioGenerateAudioData is probably coming from the input audio file. I would check your new example project, but probably it has nothing to do with the NAudio library but the file itself that's why you couldn't reproduce the issue.
